sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1494,8 +1542,11 @@ def task_worker(self):
                             self.force_cmis_reinit(lport, retries + 1)
                             continue
 
-                        # TODO: Use fine grained time when the CMIS memory map is available
-                        self.port_dict[lport]['cmis_expired'] = now + datetime.timedelta(seconds=self.CMIS_DEF_EXPIRED)
+                        if self.check_datapath_init_pending(api, host_lanes):
```

Comment:
if any of the lanes has DPInitPending=0 i.e if this fn call return False then we should retry re-init

Reply:
I have updated the code based on our discussion.